common/src/main/java/uk/gov/gchq/palisade/User.java

```diff
@@ -64,6 +65,23 @@ public void setField(final String reference, final Object value) {
         Util.setField(this, FIELD_SETTERS, reference, value);
     }
 
+    public Object getUserField(final String key) {
+        requireNonNull(key, "The key cannot be set to null.");
+        return userFields.get(key);
+    }
+
+    public void setUserField(final String key, final Object value) {
+        requireNonNull(key, "The key cannot be set to null.");
+        requireNonNull(key, "The value cannot be set to null.");
+        userFields.put(key, value);
+    }
+
+    public User userFields(final HashMap<String, Object> userFields) {
+        requireNonNull(userFields, "The userFields cannot be set to null.");
+        this.userFields = userFields;
```

Comment:
Please can this not do the assignment directly? This leaves the class open to external inconsistencies as the caller could modify the hashmap after it has been passed in. Please call clear() on the current userFields, then use putAll() to copy everything across from the one passed in.